minesweeper.py

```diff
@@ -130,8 +130,13 @@ def timer(self):
                 self.game_time += datetime.timedelta(seconds=1)
                 root.after(1000, self.timer)
             elif self.game_status == GameStatus.game_won:
-                with open("high_scores.json", "w") as high_scores:
-                    json.dump(self.game_time.strftime("%M:%S"), high_scores)
+                with open(where_this_file_is / 'high_scores.json', "w") as score_list:
+                    json_dict["high_scores"].append({'time' : self.game_time.strftime("%M:%S")}, 
+                                                     {'width' : self.width},
+                                                     {'height' : self.height},
+                                                     {'mine_count' : self.mine_count})
+                    
+                    json.dump(json_dict, score_list)
```

Comment:
Inside the append, you should have only one dict. That means one `{` and one `}`. Currently, you have four dicts: `{'time': ...}`, `{'width': ...}` and so on.

Like I said, file opening and dumping should move after `root.mainloop()`, but the append can stay here until next review.